tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_access.c

```diff
@@ -148,40 +165,112 @@ static const struct bt_mesh_model_op model_op3[] = {
 	BT_MESH_MODEL_OP_END
 };
 
+static const struct bt_mesh_model_op model_op4[] = {
+	{ TEST_MESSAGE_OP_4, 0, test_msg_handler },
+	BT_MESH_MODEL_OP_END
+};
+
+static const struct bt_mesh_model_op model_op5[] = {
+	{ TEST_MESSAGE_OP_5, 0, test_msg_handler },
+	BT_MESH_MODEL_OP_END
+};
+
+static const struct bt_mesh_model_op model_ne_op1[] = {
+	{ TEST_MESSAGE_OP_1, 0, test_msg_ne_handler },
+	BT_MESH_MODEL_OP_END
+};
+
+static const struct bt_mesh_model_op model_ne_op2[] = {
+	{ TEST_MESSAGE_OP_2, 0, test_msg_ne_handler },
+	BT_MESH_MODEL_OP_END
+};
+
+static const struct bt_mesh_model_op model_ne_op3[] = {
+	{ TEST_MESSAGE_OP_3, 0, test_msg_ne_handler },
+	BT_MESH_MODEL_OP_END
+};
+
+static const struct bt_mesh_model_op model_ne_op4[] = {
+	{ TEST_MESSAGE_OP_4, 0, test_msg_ne_handler },
+	BT_MESH_MODEL_OP_END
+};
+
+static const struct bt_mesh_model_op model_ne_op5[] = {
+	{ TEST_MESSAGE_OP_5, 0, test_msg_ne_handler },
+	BT_MESH_MODEL_OP_END
+};
+
 static struct bt_mesh_cfg_cli cfg_cli;
 
+/* do not change model sequence. it will break pointer arithmetic. */
 static struct bt_mesh_model models[] = {
 	BT_MESH_MODEL_CFG_SRV,
 	BT_MESH_MODEL_CFG_CLI(&cfg_cli),
 	BT_MESH_MODEL_CB(TEST_MODEL_ID_1, model_op1, &model_pub1, NULL, &test_model1_cb),
 	BT_MESH_MODEL_CB(TEST_MODEL_ID_2, model_op2, NULL, NULL, &test_model2_cb),
 	BT_MESH_MODEL_CB(TEST_MODEL_ID_3, model_op3, NULL, NULL, &test_model3_cb),
+	BT_MESH_MODEL_CB(TEST_MODEL_ID_4, model_op4, NULL, NULL, &test_model4_cb),
+	BT_MESH_MODEL_CB(TEST_MODEL_ID_5, model_op5, NULL, NULL, &test_model5_cb),
+};
+
+/* do not change model sequence. it will break pointer arithmetic. */
+static struct bt_mesh_model models_ne[] = {
+	BT_MESH_MODEL_CB(TEST_MODEL_ID_1, model_ne_op1, NULL, NULL, &test_model1_cb),
+	BT_MESH_MODEL_CB(TEST_MODEL_ID_2, model_ne_op2, NULL, NULL, &test_model2_cb),
+	BT_MESH_MODEL_CB(TEST_MODEL_ID_3, model_ne_op3, NULL, NULL, &test_model3_cb),
+	BT_MESH_MODEL_CB(TEST_MODEL_ID_4, model_ne_op4, NULL, NULL, &test_model4_cb),
+	BT_MESH_MODEL_CB(TEST_MODEL_ID_5, model_ne_op5, NULL, NULL, &test_model5_cb),
 };
 
 static struct bt_mesh_model vnd_models[] = {};
 
 static struct bt_mesh_elem elems[] = {
 	BT_MESH_ELEM(0, models, vnd_models),
+	BT_MESH_ELEM(1, models_ne, vnd_models),
 };
 
 const struct bt_mesh_comp local_comp = {
 	.elem = elems,
 	.elem_count = ARRAY_SIZE(elems),
 };
+/*     extension dependency
+ *         m1    m2
```

Comment:
I interpret this picture as: m1 extends m5 and m3, m2 extends m3, m3 extends m4. But it is actually the other way around. Please add a note that the top vertices are the base models for the bottom ones.